training/src/main/scala/com/airbnb/aerosolve/training/NDTree.scala

```diff
@@ -61,16 +64,33 @@ object NDTree {
     if ((indices.length > 0) && (dimensions > 0) && isConstantDimension) {
       val node = new NDTreeNode()
       nodes.append(node)
-      buildTreeRecursive(options, points, indices, nodes, node, 1, splitType)
+      buildTreeRecursive(options, points, -1, indices, nodes, node, 1, splitType)
     }
 
     val tree = new NDTree(nodes.toArray)
     tree
   }
 
+  def getNextAxis(lastAxis: Int,
+                  deltas: Array[Double],
+                  node: NDTreeNode,
+                  minLeafWidthPercentage: Double):Int = {
+    for(i <- 1 to deltas.length) {
+      val axis = (lastAxis + i) % deltas.length
+      val rootWidth = node.getMax.get(axis) - node.getMin.get(axis)
+      val width = deltas(axis)
+      if (rootWidth > 0 && width > 0 &&
+          minLeafWidthPercentage * rootWidth <= width) {
+        return axis
```

Comment:
Roughly I think you can do
```scala
      val deltas = (1 to 10).toArray
      val lastAxis: Int = 8
      deltas.indices.view.map {
        i =>
          println(s"axis for $i")
          (lastAxis + i) % deltas.length
      }.find {
        axis =>
          println(s"looking at $axis")
          val width = deltas(axis)
          width < 3
      }
```
Notice the `view` function after `indices` converts the collection to be lazy and only does map/find one element at a time as oppose to assemble the entire axis array (if this is something you are concerned about.) Otherwise I recommend you leave it out to avoid confusion.